File: docs/versioned_docs/version-1.11.0/API-Reference/python-examples/api-logs/stream-logs.py
import os

import requests

base = os.environ.get("LANGFLOW_URL") or os.environ.get("LANGFLOW_SERVER_URL", "")
api_key = os.environ.get("LANGFLOW_API_KEY", "")

# `/logs-stream` is an SSE endpoint. For doc example stability, only read a small
# number of events, then close the connection.
url = f"{base}/logs-stream"
headers = {"accept": "text/event-stream", "x-api-key": api_key}

response = requests.get(url, headers=headers, stream=True, timeout=30)
response.raise_for_status()

events_read = 0
chunks: list[str] = []
for line in response.iter_lines(decode_unicode=True):
    if line:
        chunks.append(line)
        events_read += 1
        if events_read >= 3:
            break

response.close()

print("\n".join(chunks))

File: docs/versioned_docs/version-1.12.0/API-Reference/javascript-examples/api-logs/stream-logs.js
const url = `${process.env.LANGFLOW_URL ?? ""}/logs-stream`;

const options = {
  method: 'GET',
  headers: {
    "accept": `text/event-stream`,
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));
